projects/plugins/jetpack/_inc/client/my-plan/my-plan-banner/index.jsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import React, { useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'lib/analytics';
+import MyPlanCard from '../my-plan-card';
+import { imagePath } from 'constants/urls';
+
+/**
+ * Import styles
+ */
+import './style.scss';
+
+const MyPlanBanner = props => {
+	const { productSlug, action, title, tagLine, trackingId } = props;
+
+	useEffect( () => {
+		analytics.tracks.recordEvent( 'jetpack_my_plan_banner_view', {
+			type: trackingId,
+		} );
+	}, [ trackingId ] );
+
+	const trackActionClick = useCallback( () => {
+		analytics.tracks.recordJetpackClick( {
+			target: trackingId,
+			feature: 'my-plan-banner',
+			page: 'my-plan',
+		} );
```

Comment:
Wouldn't it interesting to have the coupon name in there, so we could separate banner views and banner clicks per partner?

Reply:
Thank you for the suggestion Jeremy!
Just to close the circle: https://github.com/Automattic/jetpack/pull/22705